apps/stripe/src/app/api/saleor/payment-gateway-initialize-session/use-case-response.ts

```diff
@@ -0,0 +1,30 @@
+import { buildSyncWebhookResponsePayload } from "@saleor/app-sdk/handlers/shared";
+
+import { SuccessWebhookResponse } from "@/modules/saleor/saleor-webhook-responses";
+
+import { ResponseDataType } from "./response-params";
+
+class Success extends SuccessWebhookResponse {
+  responseData: ResponseDataType;
+
+  constructor(args: { responseData: ResponseDataType }) {
```

Comment:
I think `ResponseDataType` should be part ot "UseCaseResponse".

IMO you should do new `Success({pk: PublishableKey})` and success knows internally (getResponse) how to map PublshableKey to this specific ResponseDataType

In you implementation, you require UseCase to "know" the data structure of the response, even if you proposed abstraction for it already (SuccessWebhookResponse)

In this example this is not that problematic, but once we have more complex examples, your use-case will be unnecessarily bloated